hvantk/ptm/pipeline.py

```
@@ -121,14 +123,26 @@ def download_ensembl_gtf(output_dir: str, overwrite: bool = False) -> str:
+    host = parsed.hostname or ""
+    if host != "ftp.ensembl.org" and not host.endswith(".ensembl.org"):
+        raise ValueError(
+            "Invalid Ensembl GTF URL host "
+            "(expected trusted Ensembl host, e.g. ftp.ensembl.org or *.ensembl.org): %s"
+            % ENSEMBL_GTF_URL
+        )
 
-    urlretrieve(ENSEMBL_GTF_URL, gtf_path)
-    logger.info(f"Downloaded: {gtf_path}")
+    logger.info("Downloading Ensembl GTF to %s...", gtf_path)
+    urllib.request.urlretrieve(ENSEMBL_GTF_URL, gtf_path)  # nosec B310
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)